ras/kprobe.py

```diff
@@ -0,0 +1,135 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM.
+# Author: Kamalesh Babulal <kamalesh@linux.vnet.ibm.com>
+
+import os
+import shutil
+import tempfile
+from avocado import Test
+from avocado import main
+from avocado.utils import distro
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Kprobe(Test):
+
+    """
+    Test kernel kprobe
+    :avocado: tags=privileged
+    """
+
+    fail_cmd = list()
+
+    def run_cmd(self, cmd):
+        self.log.info("executing ============== %s =================", cmd)
+        if process.system(cmd, ignore_status=True, sudo=True, shell=True):
+            self.is_fail += 1
+            self.fail_cmd.append(cmd)
+        return
+
+    @staticmethod
+    def run_cmd_out(cmd):
+        return process.system_output(cmd, shell=True, ignore_status=True,
+                                     sudo=True)
+
+    def clear_dmesg(self):
+        process.run("dmesg -C ", sudo=True)
+
+    def setUp(self):
+        """
+        Setting up the env for the kernel building
+        """
+        smg = SoftwareManager()
+        detected_distro = distro.detect()
+        deps = ['gcc', 'make', 'automake', 'autoconf', 'time', 'bison', 'flex']
+        if 'Ubuntu' in detected_distro.name:
+            linux_headers = 'linux-headers' + os.uname()[2]
+            deps.extend(['libpopt0', 'libc6', 'libc6-dev', 'libpopt-dev',
+                         'libcap-ng0', 'libcap-ng-dev', 'elfutils', 'libelf1',
+                         'libnuma-dev', 'libfuse-dev', 'libssl-dev', linux_headers])
+        elif 'SuSE' in detected_distro.name:
+            deps.extend(['libpopt0', 'glibc', 'glibc-devel',
+                         'popt-devel', 'libcap2', 'libcap-devel', 'kernel-symm',
+                         'libcap-ng-devel', 'openssl-devel', 'kernel-source'])
+        elif detected_distro.name in ['centos', 'fedora', 'rhel']:
+            deps.extend(['popt', 'glibc', 'glibc-devel', 'libcap-ng',
+                         'libcap', 'libcap-devel', 'elfutils-libelf',
+                         'elfutils-libelf-devel', 'openssl-devel',
+                         'kernel-devel', 'kernel-headers'])
+        for package in deps:
+            if not smg.check_installed(package) and not smg.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+    def build_module(self):
+        """
+        Building of the kprobe kernel module
+        """
+        self.log.info("============== Building kprobe Module =================")
+        self.sourcedir = tempfile.mkdtemp()
+        os.chdir(self.sourcedir)
+
+        self.location = ('https://raw.githubusercontent.com/torvalds/linux/'
+                         'master/samples/kprobes/kprobe_example.c')
+        kprobe_file = self.fetch_asset(self.location, expire='1d')
```

Comment:
is this files changes every day ?

Reply:
I choosed that value based on the `kernbench.py` but yeah on other hand it might not change everyday, So do you think `7days` as an ideal number ?

Reply:
generally we use 7 days for all the place can be good option if any other thought is welcome , as it just make sure to avoid download if there is no change in file